contracts/src/TroveManager.sol

```diff
@@ -1263,7 +1264,8 @@ contract TroveManager is LiquityBase, ITroveManager, ITroveEvents {
         // Push the trove's id to the Trove list
         TroveIds.push(_troveId);
 
-        _updateBatchShares(_troveId, _batchAddress, _troveChange, _batchColl, _batchDebt, true);
+        assert(_troveChange.debtIncrease > 0); // TODO: remove before deployment
```

Comment:
Actually, I wondered if we could have fixed the original issue with a check that batch debt always increases by the same amount as the caller's BOLD balance (or, at least within some low tolerance). e.g. enforcing the core borrowing invariant.

Though I think we'd still need this PR for redistributions.